parser.py

```diff
@@ -48,6 +48,10 @@ def __init__(self):
 
         self.p.add_argument('--inference-passes', '-i', type=int, default=10,
                 help='number of forward passes during test')
+        self.p.add_argument('--training-passes', type=int, default=1)
+        self.p.add_argument('--val-passes', '-v', type=int, default=1)
+        self.p.add_argument('--val-gumbel', action='store_true',
```

Comment:
Commenting here because other file isnt in pr, we should have option to switch test on and off too